test/e2e/framework/syncer.go

```diff
@@ -503,6 +523,64 @@ func (sf *syncerFixture) Start(t *testing.T) *StartedSyncerFixture {
 		SyncerVirtualWorkspaceConfig:   syncerVWConfig,
 		UpsyncerVirtualWorkspaceConfig: upsyncerVWConfig,
 	}
+}
+
+// SimulateStartWithoutSyncing simulates the start of a Syncer and performs what the Syncer does to bring and maitain
+// the SyncTarget to a Ready status (heartbeat, and API schema compatibility).
+// No resource will be without effectively synced after calling this method.
+func (sf *syncerFixture) SimulateStartWithoutSyncing(t *testing.T) *StartedSyncerFixture {
+	t.Helper()
+
+	artifactDir, _, err := ScratchDirs(t)
+	if err != nil {
+		t.Errorf("failed to create temp dir for syncer artifacts: %v", err)
+	}
+
+	downstreamConfig, _, downstreamKubeClient, syncerConfig, syncerID := sf.createAndApplySyncTarget(t, false, artifactDir)
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	t.Cleanup(cancelFunc)
+
+	kcpBootstrapClusterClient, err := kcpclientset.NewForConfig(syncerConfig.UpstreamConfig)
+	require.NoError(t, err)
+	kcpSyncTargetClient := kcpBootstrapClusterClient.Cluster(syncerConfig.SyncTargetPath)
+
+	// Artificially consider that all the resources to sync are authorized on the physical cluster side.
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		syncTarget, err := kcpSyncTargetClient.WorkloadV1alpha1().SyncTargets().Get(ctx, syncerConfig.SyncTargetName, metav1.GetOptions{})
+		require.NoError(t, err)
+		conditions.Set(syncTarget, conditions.TrueCondition(workloadv1alpha1.SyncerAuthorized))
+		_, err = kcpSyncTargetClient.WorkloadV1alpha1().SyncTargets().UpdateStatus(ctx, syncTarget, metav1.UpdateOptions{})
+		return err
+	})
+	require.NoError(t, err)
+
+	// Import the resource schemas of the resources to sync from the physical cludster, to enable compatibility check in KCP.
+	resources := syncerConfig.ResourcesToSync.List()
+	kcpSyncTargetInformerFactory := kcpinformers.NewSharedScopedInformerFactoryWithOptions(kcpSyncTargetClient, 10*time.Hour, kcpinformers.WithTweakListOptions(
+		func(listOptions *metav1.ListOptions) {
+			listOptions.FieldSelector = fields.OneTermEqualSelector("metadata.name", syncerConfig.SyncTargetName).String()
+		},
+	))
+	kcpImporterInformerFactory := kcpinformers.NewSharedScopedInformerFactoryWithOptions(kcpSyncTargetClient, 10*time.Hour)
+	apiImporter, err := syncer.NewAPIImporter(
+		syncerConfig.UpstreamConfig, syncerConfig.DownstreamConfig,
+		kcpSyncTargetInformerFactory.Workload().V1alpha1().SyncTargets(),
+		kcpImporterInformerFactory.Apiresource().V1alpha1().APIResourceImports(),
+		resources,
+		syncerConfig.SyncTargetPath, syncerConfig.SyncTargetName, types.UID(syncerConfig.SyncTargetUID))
+	require.NoError(t, err)
+
+	kcpImporterInformerFactory.Start(ctx.Done())
+	kcpSyncTargetInformerFactory.Start(ctx.Done())
+	kcpSyncTargetInformerFactory.WaitForCacheSync(ctx.Done())
+
+	go apiImporter.Start(klog.NewContext(ctx, klog.FromContext(ctx).WithValues("resources", resources)), 5*time.Second)
```

Comment:
why is the api importer needed in this fixture?

Could we make it optional in another `StartAPIImporter` method?

I.e. something like `syncerFxture.StartHeartBeat(...).StartAPIImport(...)`?

Reply:
Rule of thumb: smaller fixtures

Reply:
And a fixture without the need of a downstream cluster would be probably a good idea.

Reply:
For the SyncTarget to be ready to use, we would expected, in most tests, that the API compatibility check has been run and that the `Accepted` (or `Imcompatible` in some cases) state would be set to every resource of the `synctarget.Status.SyncedResources` array.

There are very few e2e tests that don't even require this information in the SyncTarget, even maybe none.

Reply:
I have separated the 2 functions as you proposed:

```
syncerFixture.CreateAndStartHeartBeat(t).StartAPIImporter(t)
```

>And a fixture without the need of a downstream cluster would be probably a good idea.

I'd rather keep that for later, since I don't see the need of this right now, and this would involve more coding / refactoring. Do you agree ?

Reply:
Thanks! worked perfectly for my PR! being able to start/stop the heartbeat allows us to test some "complex" scheduling scenarios!